src/tools/perf/perftest_context.h

```diff
@@ -60,6 +60,39 @@ struct perftest_context {
 };
 
 
+typedef struct perftest_params_local_backup {
+    char                 dev_name[UCT_DEVICE_NAME_MAX];
+    uint32_t             needed_flags;
+} perftest_params_local_backup_t;
+
+
+static inline void
+perftest_params_local_backup(const perftest_params_t *params,
+                             perftest_params_local_backup_t *backup)
+{
+    if (params->super.api == UCX_PERF_API_UCT) {
+        strcpy(backup->dev_name, params->super.uct.dev_name);
+    } else {
+        backup->dev_name[0] = '\0';
+    }
+
+    backup->needed_flags = params->super.flags &
+                           UCX_PERF_TEST_FLAG_ERR_HANDLING;
+}
+
+
+static inline void
+perftest_params_local_restore(perftest_params_t *params,
+                              const perftest_params_local_backup_t *backup)
+{
+    if (params->super.api == UCX_PERF_API_UCT) {
```

Comment:
for my understanding, are we sure this `UCX_PERF_API_UCT` value is not remove from receiving params content? would it make sense to also embed it in `backup` somehow?

Reply:
I don't think that it's legal and robust to run server with API_UCT parameters but client with API_UCP ones, probably we should add such test and related error handling, but this is out of scope of this PR